### ISSUE TITLE
Gradient leakage from test dataset of the TGIF_QA

### ISSUE THREAD

**Issue by Fly2flies** [salesforce/BiST#2]
Hello,
Thank you for sharing the great work and code base! 
But I noticed that your TGIF_QA training is unreasonable, causing gradient leakage and ultimately resulting in false over-accuracy on the test set.

Problem is : the gradient on the test set is not cleared during each epoch test, and it will accumulate to the next training [epoch.]  [optimizer code](https://github.com/salesforce/BiST/blob/dadd9070c580652bab518a68156fb4eea6fb7415/model/optimize.py#L84)

The usual practice is to wrap it with `torch.no_grad` while validating or testing, or call `optimizer.zero_grad` first before `loss.backward`.

Maybe you can check your VideoQA experiment again, because your performance is very high.

**Comment by henryhungle** [comment]
Hi @Fly2flies thank you for noting the error. 

I checked the code and noticed it was an old version of the code. I updated the code to fix the error with `torch.no_grad` and will check the results again.
